bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3Storage.java

```diff
@@ -263,18 +262,10 @@ private StreamSegmentInformation doGetStreamSegmentInfo(String streamSegmentName
 
     private boolean doExists(String streamSegmentName) {
         try {
-            ListObjectsResult result = client.listObjects(config.getBucket(), config.getPrefix() + streamSegmentName);
-            return !result.getObjects().isEmpty();
+            S3ObjectMetadata result = client.getObjectMetadata(config.getBucket(),
+                    config.getPrefix() + streamSegmentName);
```

Comment:
_streamSegmentName_ is actually a stream segment name but do we actually create objects with such names? I thought   for each segment there will be objects like _streamSegmentName$header_ or _streamSegmentName$attributes_. So if we simple check for presence of an object with _streamSegmentName_ we always get object not found. Am i missing something here?

Reply:
The layer above this component (RollingStorage) passes the names like _streamSegmentName$header_ or _streamSegmentName$attributes_ to this component.